classes/plugins/ThemePlugin.inc.php

```diff
@@ -821,6 +824,57 @@ function isColourDark( $colour, $limit = 130 ) {
 		);
 		return $contrast < $limit;
 	}
+	
+	/**
+	 * Add default stylesheets to the HTML galley
+	 *
+	 * Passes styles to the galley through HTML Galley Plugin hook
+	 *
+	 * @param $hookName string
+	 * @param $args array [
+	 * $args[0] ArticleGalley
+	 * $args[1] string HTML
+	 * ]
+	 *
+	 * @return bool
+	 */
+	function addHmtlGalleyStyle($hookName, $args) {
+		$galley =& $args[0];
+		$contents =& $args[1];
+		
+		if (empty($contents)) return false;
+		
+		$submissionFile = $galley->getFile();
+		
+		// Check if HTML galley has attached styles
+		$submissionFileDao = DAORegistry::getDAO('SubmissionFileDAO');
+		import('lib.pkp.classes.submission.SubmissionFile');
+		$embeddableFiles = array_merge(
+			$submissionFileDao->getLatestRevisions($submissionFile->getSubmissionId(), SUBMISSION_FILE_PROOF),
+			$submissionFileDao->getLatestRevisionsByAssocId(ASSOC_TYPE_SUBMISSION_FILE, $submissionFile->getFileId(), $submissionFile->getSubmissionId(), SUBMISSION_FILE_DEPENDENT)
+		);
+		
+		$attachedStyles = false;
+		
+		foreach ($embeddableFiles as $embeddableFile) {
+			if ($embeddableFile->getFileType() =='text/css') $attachedStyles = true;
+		}
+		
+		if ($attachedStyles) return false;
+		
+		// Upload styles to the HTML galley
+		$output = '';
+		
+		foreach ($this->styles as $styleData) {
```

Comment:
Where do these come from? I'm confused about how a `ThemePlugin` will register a default CSS file. Also, in this case, a LESS file can't be used. 

Is there any way we can use our LESS compiler to support a LESS file as well? I imagine I'd want to reuse the variables from my theme's LESS stylesheets in the HTML galley as well.

Reply:
Styles in the `ThemePlugin` are passing to the `PKPTemplateManager` through `_registerStyles` method but here I'm retrieving them directly as they actually won't go to the template; they are injected directly into the HTML. Probably, I need to ensure they that are beeing removed from `$styles` property after beeing used and not passing further to the Template Manager. Or you have something different in mind?

I think it is not a problem to add a call to the compile library in t his method for LESS files if the overall approach looks reasonable.

Reply:
Ah ok, so if I'm reading this properly, a stylesheet is added in a theme like this:

```
$this->addStyle('name', 'path/to/file.css', ['contexts' => 'htmlGalley']);
```

Is that right? If so, then LESS files should already be handled. `_registerStyles` will transform the URL appropriately.

You'll need to handle the case where `contexts` is an array. `TemplateManager::getResourcesByContext($resources, $context)` can do this for you. Usage would be like this:

```
$templateMgr = TemplateManager::getManager($request);
$styles = $templateMgr->getResourcesByContext($this->styles, 'htmlGalley');
```

That will get styles with any context that includes `htmlGalley`. :+1:

Reply:
Yes, right. But the method retrieves the data from styles earlier, before they are registered and passed to Template Manager's methods.